hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/RDBStore.java

```diff
@@ -104,6 +107,17 @@ public RDBStore(File dbFile, ManagedDBOptions dbOptions,
         }
       }
 
+      //create snapshot directory if does not exist.
+      snapshotsParentDir = Paths.get(dbLocation.getParent(),
+          OM_SNAPSHOT_DIR).toString();
+      File snapshotsDir = new File(snapshotsParentDir);
+      if (!snapshotsDir.exists()) {
+        boolean success = snapshotsDir.mkdir();
+        if (!success) {
+          LOG.warn("Unable to create RocksDB snapshot directory");
```

Comment:
Shouldn't we throw an exception here ?

Reply:
I was just imitating the code above that creates the checkpoint directory: https://github.com/apache/ozone/blob/d551aa4635e4d33b0080d46d49d8d5b26aead820/hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/RDBStore.java#L99-L108

Reply:
We shud bail out early here, warning is not enough.